apps/www/registry/default/ui/calendar.tsx

```diff
@@ -1,62 +1,312 @@
+        button_next: cn(
+          buttonVariants({
+            variant: "outline",
+            className:
+              "absolute right-0 h-7 w-7 bg-transparent p-0 opacity-50 hover:opacity-100",
+          })
         ),
-        nav_button_previous: "absolute left-1",
-        nav_button_next: "absolute right-1",
-        table: "w-full border-collapse space-y-1",
-        head_row: "flex",
-        head_cell:
-          "text-muted-foreground rounded-md w-9 font-normal text-[0.8rem]",
-        row: "flex w-full mt-2",
-        cell: "h-9 w-9 text-center text-sm p-0 relative [&:has([aria-selected].day-range-end)]:rounded-r-md [&:has([aria-selected].day-outside)]:bg-accent/50 [&:has([aria-selected])]:bg-accent first:[&:has([aria-selected])]:rounded-l-md last:[&:has([aria-selected])]:rounded-r-md focus-within:relative focus-within:z-20",
-        day: cn(
+        button_previous: cn(
+          buttonVariants({
+            variant: "outline",
+            className:
+              "absolute left-0 h-7 w-7 bg-transparent p-0 opacity-50 hover:opacity-100",
+          })
+        ),
+        nav: "flex items-start",
+        month_grid: "mt-4",
+        week: "mt-2 flex w-full",
+        day: "flex size-8 flex-1 items-center justify-center rounded-md p-0 text-sm [&:has(button)]:hover:!bg-accent [&:has(button)]:hover:text-accent-foreground [&:has(button)]:hover:aria-selected:!bg-primary [&:has(button)]:hover:aria-selected:text-primary-foreground",
+        day_button: cn(
           buttonVariants({ variant: "ghost" }),
-          "h-9 w-9 p-0 font-normal aria-selected:opacity-100"
+          "h-8 w-8 p-0 font-normal transition-none hover:bg-transparent hover:text-inherit aria-selected:opacity-100"
         ),
-        day_range_end: "day-range-end",
-        day_selected:
-          "bg-primary text-primary-foreground hover:bg-primary hover:text-primary-foreground focus:bg-primary focus:text-primary-foreground",
-        day_today: "bg-accent text-accent-foreground",
-        day_outside:
+        range_start: "day-range-start rounded-s-md",
+        range_end: "day-range-end rounded-e-md",
+        selected:
+          "bg-primary text-primary-foreground hover:!bg-primary hover:text-primary-foreground focus:bg-primary focus:text-primary-foreground",
+        today: "bg-accent text-accent-foreground",
+        outside:
           "day-outside text-muted-foreground opacity-50 aria-selected:bg-accent/50 aria-selected:text-muted-foreground aria-selected:opacity-30",
-        day_disabled: "text-muted-foreground opacity-50",
-        day_range_middle:
-          "aria-selected:bg-accent aria-selected:text-accent-foreground",
-        day_hidden: "invisible",
+        disabled: "text-muted-foreground opacity-50",
+        range_middle:
+          "rounded-none aria-selected:bg-accent aria-selected:text-accent-foreground hover:aria-selected:!bg-accent hover:aria-selected:text-accent-foreground",
+        hidden: "invisible hidden",
```

Comment:
I'm observing that when showOutsideDays is false for a 2 month calendar, the "hidden" property messes with the flex positioning since the hidden days still need to take up space along the row

Reply:
Same here. The `hidden` class should be removed